jax/_src/lru_cache.py

```diff
@@ -153,26 +169,38 @@ def _evict_if_needed(self, *, additional_size: int = 0) -> None:
     if not self.eviction_enabled:
       return
 
-    # a priority queue, each element is a tuple `(file_mtime, file, file_size)`
-    h: list[tuple[int, pathlib.Path, int]] = []
+    # a priority queue, each element is a tuple `(file_atime, key, file_size)`
+    h: list[tuple[int, str, int]] = []
     dir_size = 0
-    for file in self.path.iterdir():
-      if file.is_file() and file != self.lock_path:
-        file_size = file.stat().st_size
-        file_mtime = file.stat().st_mtime_ns
+    for cache_file in self.path.glob(f"*{_CACHE_SUFFIX}"):
+      if not cache_file.is_file():
+        continue
 
-        dir_size += file_size
-        heapq.heappush(h, (file_mtime, file, file_size))
+      file_size = cache_file.stat().st_size
+
+      key = cache_file.name.removesuffix(_CACHE_SUFFIX)
+      atime_file = self.path / f"{key}{_ATIME_SUFFIX}"
+      file_atime = int.from_bytes(atime_file.read_bytes(), "little")
+
+      dir_size += file_size
+      heapq.heappush(h, (file_atime, key, file_size))
 
     target_size = self.max_size - additional_size
     # evict files until the directory size is less than or equal
     # to `target_size`
     while dir_size > target_size:
-      file_mtime, file, file_size = heapq.heappop(h)
-      msg = (f"Evicting cache file {file.name}: file size {file_size} bytes, "
+      file_atime, key, file_size = heapq.heappop(h)
+
+      msg = (f"Evicting cache entry {key!r}: file size {file_size} bytes, "
```

Comment:
Inline `msg` into the logger call? You could also do

```
logger.debug("Evicting cache entry {key!r} ...", key=key)
```

to avoid string formatting if logging is disabled.

Reply:
I tried this syntax but it does not work:

```python
import logging
logger = logging.getLogger(__name__)
logger.critical("Evicting cache entry {key1!r}: file size {file_size} bytes, target cache size {target_size} bytes", key1="123", file_size=123, target_size=123)
```

Got this error:

```
TypeError: Logger._log() got an unexpected keyword argument 'key'
```

Should that be

```python
logger.debug("Evicting cache entry %r: file size %d bytes, target cache size %d bytes", key, file_size, target_size)
```

instead?

Reply:
Hmm, I thought loggers support both, but if the latter works, use that, yeah.